packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -352,15 +377,31 @@ export function jsxProdSignatureRunningInDevWithDynamicChildren(
     const trackActualOwner =
       __DEV__ &&
       ReactSharedInternals.recentlyCreatedOwnerStacks++ < ownerStackLimit;
+    let debugStackDEV = false;
+    if (__DEV__) {
+      if (trackActualOwner) {
+        if (supportsSettingStackTraceLimit) {
+          try {
+            const previousStackTraceLimit = Error.stackTraceLimit;
+            Error.stackTraceLimit = ownerStackTraceLimit;
+            debugStackDEV = Error('react-stack-top-frame');
+            Error.stackTraceLimit = previousStackTraceLimit;
+          } catch {}
```

Comment:
try/catch is notorious for deopting. It's better in modern browsers but can still trigger extra stuff on the stack and avoid inlining capabilities by making larger code.

We need to cut everything. We should just let it throw if this is not supported.